compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2396,6 +2396,11 @@ object Parsers {
         }
         else Nil
       }
+
+      if (ofCaseClass && in.token != LPAREN)
+        syntaxError( "case classes without a parameter list are not allowed;\n"+
+          "use either case objects or case classes with an explicit `()' as a parameter list.", in.lastOffset)
+
       recur(firstClause = true, 0, ofInstance)
```

Comment:
I would do something along the lines of:

```scala
val start = in.offset
val paramss = recur(firstClause = true, 0, ofInstance)

if (ofCaseClass && paramss.isEmpty)
  syntaxError(..., start)
```

Reply:
makes sense. Thank you :+1: